src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -634,6 +635,12 @@ private String lastDottedPart(String input) {
     private final Set<String> provides;
     private int valueSymbolsWalked = 0;
     private final boolean isExtern;
+    private final Predicate<JSType> isUnknownType = new Predicate<JSType>() {
```

Comment:
The naming is a bit odd - unknown vs undefined vs void vs optional. Maybe call this `isVoidType`, and add a comment on why void means optional in unions when translated to TS?